src/firefly/js/drawingLayers/RegionPlot.js

```diff
@@ -319,4 +392,45 @@ function removeRegionsFromData(drawLayer, removedRegions) {
     drawLayer.regions = resultRegions;
     drawLayer.drawObjAry = allDrawObjs;
     return allDrawObjs;
+}
+
+/**
+ * @summary find the region drawObj based on region description
+ * @param {object} drawLayer
+ * @param {string|string[]} regionDes
+ * @param {int} stopIndex maximum number of regions to be selected
+ * @return {Object}
+ * @ignore
+ */
+function getSelectedRegionDrawObj(drawLayer, regionDes, stopIndex = 1) {
+    var {regions, drawObjAry} = drawLayer;
+
+    var regs = RegionFactory.parseRegionDS9((isString(regionDes) ? [regionDes] : regionDes),
+                                            true, stopIndex);
+    var selDrawObj = [];
+
+    if (!isEmpty(regs)) {
+        selDrawObj = regs.reduce((prev, aRegion, index) => {
+            if (index < stopIndex) {
+                var idx = getRegionIndex(regions, aRegion);
+
+                if (idx >= 0) {
+                    prev.push(drawObjAry[idx]);
+                }
+            }
+            return prev;
+        }, []);
+    }
+    return selDrawObj.slice(0, stopIndex);
+}
+
+/**
+ * @summary get the region description of the selected region from the specified region layer
+ * @param {string} drawLayerId id of the drawing layer
+ * @return {string}
```

Comment:
if the click occurs nowhere (no region under), then the previous selected region will becomes deselected. Passing a 'null' is the way to deselect the region.